avocado/core/runner.py

```diff
@@ -239,7 +239,8 @@ def finish(self, proc, started, timeout, step):
                        test_state['name'])
         if proc.is_alive():
             TEST_LOG.warning("Killing hanged test process %s" % proc.pid)
-            for _ in xrange(5):     # I really want to destroy it
+            end_time = time.time() + 60
+            while time.time() < end_time:
```

Comment:
PS: I know only of one example where killing takes longer than 5s and that is removing and re-creating a big file on `xfs` and there just one kill suffices, still it takes up to several minutes to finish...